kube-client/src/api/util/mod.rs

```diff
@@ -59,6 +66,56 @@ impl Api<ServiceAccount> {
     }
 }
 
+/// Updates stored versions of all objects so that is matches
+/// stored version of the CustomResourceDefinition.
+///
+/// This function makes apiserver
+/// migrate all its instances to the version, marked with `storage=true`, and deletes all other versions
+/// from the `.status.storedVersions`.
+/// # Guarantees
+/// - This function never corrupts or loses data and does not affect API clients (assuming that conversion is set up properly)
+/// - If this function failed (or was aborted), you can safely call it again.
+/// - If this function succeeded, you can safely call it again (but you shouldn't - this is useless)
+/// - While this function runs, you can call safely it again in parallel (but you shouldn't - this is useless)
+/// - If this function succeeded, you may safely stop serving any other api versions
+pub async fn migrate_resources<K: Resource + Clone + Serialize + DeserializeOwned + std::fmt::Debug>(
+    api: Api<K>,
```

Comment:
I added a function to go from CRD to `ApiResource` for its instances.
Unfortunately, this conversion is not one-to-one, because CRD with multiple versions maps to multiple API resource. Therefore I had to add additional hint parameter.

However, it still greatly simplified `migrate_resources` signature and implementation/

Reply:
Looks much better!